release/ray_release/test.py

```diff
@@ -83,6 +123,65 @@ def get_anyscale_byod_image(self) -> str:
         """
         return f"{DATAPLANE_ECR}/{self.get_byod_repo()}:{self.get_byod_image_tag()}"
 
+    def add_test_result(self, result: Result) -> None:
+        """
+        Add test result to test object
+        """
+        self.get_test_results().insert(0, TestResult.from_result(result))
+
+    def get_test_results(self, limit: int = 10) -> List[TestResult]:
+        """
+        Get test result from test object, or s3
+        """
+        if self.test_results is not None:
+            return self.test_results
```

Comment:
why is this cached? is this just for testing? or this is actually required for performance?

Reply:
mostly for performance, since the incoming of test result is sparse, reloading in one request might be wasteful

i can also add a 'refresh' flag or something along that line as an option